app/assets/javascripts/components/admin/unmatched_page.es6.jsx

```diff
@@ -0,0 +1,190 @@
+class UnmatchedPage extends React.Component {
+
+  constructor(props) {
+    super();
+    this.state = {
+      student_detail: false,
+      teacher_detail: false,
+      students: null,
+      teachers: null,
+      student: null,
+      teacher: null,
+    }
+
+    this.studentOnClick = this.studentOnClick.bind(this);
+    this.teacherOnClick = this.teacherOnClick.bind(this);
+    this.studentBackButton = this.studentBackButton.bind(this);
+    this.teacherBackButton = this.teacherBackButton.bind(this);
+    this.makeMatching = this.makeMatching.bind(this);
+  }
+
+  componentDidMount() {
+    this.loadStudents()
+  }
+
+  loadStudents() {
+    var studentRoute = ApiConstants.students.unmatched;
+    var studentResolve = (response) => this.setState({students: response["students"]});
+    var studentReject = (response) => console.log(response);
+
+    Requester.get(
+      studentRoute,
+      studentResolve,
+      studentReject,
+    );
+  }
+
+  studentOnClick(student_id) {
+    var studentRoute = ApiConstants.students.show(student_id)
+
+    var studentResolve = ((response) => {
+      this.setState({student_detail: true, student: response});
+
+      var teacherRoute = ApiConstants.teachers.possible_teachers(student_id);
+      var teacherResolve = (response) => this.setState({teachers: response["teachers"]});
+      var teacherReject = (response) => console.log(response);
+
+      Requester.get(
+        teacherRoute,
+        teacherResolve,
+        teacherReject,
+      );
+    }).bind(this);
```

Comment:
when you call onclick = this.studentOnClick 
just do onClick = {() => this.studentOnClick()} so you don't have to bind the scope for anything